src/screens/Settings/components/SyncSaves/gog.tsx

```diff
@@ -0,0 +1,251 @@
+import { useTranslation } from 'react-i18next'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons'
+
+interface Props {
+  appName: string
+  gogSaves: GOGCloudSavesLocation[]
+  setGogSaves: (saves: GOGCloudSavesLocation[]) => void
+  autoSyncSaves: boolean
+  setAutoSyncSaves: (value: boolean) => void
+}
+
+export default function GOGSyncSaves({
+  appName,
+  gogSaves,
+  setGogSaves,
+  autoSyncSaves,
+  setAutoSyncSaves
+}: Props) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [isSyncing, setIsSyncing] = useState(false)
+  const [syncType, setSyncType] = useState('--skip-upload')
+
+  const { t } = useTranslation()
+
+  const { platform } = useContext(ContextProvider)
+  const isWin = platform === 'win32'
+  const isMac = ['osx', 'Mac']
+
+  useEffect(() => {
+    const getLocations = async () => {
+      const {
+        gog_save_location,
+        install: { platform: installed_platform, install_path }
+      } = await getGameInfo(appName, 'gog')
+      const clientId = await ipcRenderer.invoke('getGOGGameClientId', appName)
+      if (!gog_save_location) {
+        return
+      }
+      setIsLoading(true)
+      const locations = gogSaves
+      if (gog_save_location?.length === 0) {
+        gog_save_location.push({
+          name: '__default',
+          location:
+            installed_platform === 'windows'
+              ? `%LocalAppData%/GOG.com/Galaxy/Applications/${clientId}/Storage/Shared/Files`
+              : `$HOME/Library/Application Support/GOG.com/Galaxy/Applications/${clientId}/Storage`
+        })
+      }
+
+      const isMacNative = isMac.includes(installed_platform ?? '')
+      const isLinuxNative = installed_platform === 'linux'
+      const isNative = isWin || isMacNative || isLinuxNative
+
+      for (const loc of gog_save_location) {
+        const { name, location } = loc
+        const saveLocation = await fixGogSaveFolder(
+          location.replace('<?INSTALL?>', String(install_path)),
+          String(installed_platform)
+        )
+        console.log(saveLocation)
+        let actualPath: string
+        if (!isNative) {
+          const { stdout } = await ipcRenderer
+            .invoke('runWineCommandForGame', {
+              appName,
+              runner: 'gog',
+              command: `cmd /c winepath "${saveLocation}"`
+            })
+            .catch((error) => {
+              console.error('There was an error getting the path', error)
+              setIsLoading(false)
+            })
+          actualPath = stdout.trim()
+        } else {
+          actualPath = await ipcRenderer.invoke('getShellPath', saveLocation)
+        }
+        const locationIndex = locations.findIndex(
+          (value) => value.name === name
+        )
+
+        console.log(actualPath)
+        if (locationIndex >= 0 && !locations[locationIndex]?.location.length) {
```

Comment:
It might be good to have `!locations[locationIndex]?.location.length` in an extra variable, just for code clarity